### PR TITLE
[IngestionClient] Add transientFailureException and retry if it was thrown

### PR DESCRIPTION
**Pull request by HenryvanderVegte**
## Purpose
Adding a new exception kind to trigger a re-enqueueing of service bus message if it was thrown to make sure that jobs don't fail because of a transient issue

## Pull Request Type
What kind of change does this Pull Request introduce?
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: samples/ingestion/ingestion-client/Connector/TransientFailureException.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+// </copyright>
+
+namespace Connector
+{
+    using System;
+
+    /// <summary>
+    /// Exception that is being thrown when azure functions should trigger a retry (up to a max. retry limit)
+    /// after a certain number of minutes.
```
Comment (mahilleb-msft) [inline samples/ingestion/ingestion-client/Connector/TransientFailureException.cs:12]: I guess it's seconds rather than minutes for you?

**Review by mahilleb-msft — APPROVED** [review]
:shipit:

File: samples/ingestion/ingestion-client/Connector/TransientFailureException.cs (hunk 1 above)
Comment (HenryvanderVegte) [reply samples/ingestion/ingestion-client/Connector/TransientFailureException.cs:12]: no, it is actually minutes (either 1,2,4... and so on) :)

**Review by mahilleb-msft — DISMISSED** [review]
:shipit:

**Review by mahilleb-msft — APPROVED** [review]

**Pull request merged**